src/cs50/sql.py

```diff
@@ -32,12 +33,23 @@ def __init__(self, url, **kwargs):
             if not os.path.isfile(matches.group(1)):
                 raise RuntimeError("not a file: {}".format(matches.group(1)))
 
-        # Create engine, raising exception if back end's module not installed
-        self.engine = sqlalchemy.create_engine(url, **kwargs)
+            foreign_keys_enabled = kwargs.pop("foreign_keys_enabled", False)
+
+            # Create engine, raising exception if back end's module not installed
+            self.engine = sqlalchemy.create_engine(url, **kwargs)
+
+            # Whether to enable foreign key constraints
+            if foreign_keys_enabled:
+                sqlalchemy.event.listen(self.engine, "connect", _on_connect)
+        else:
```

Comment:
Does PEP require a blank line above comment? Not sure.

Reply:
I thought that was your convention? :stuck_out_tongue:

Reply:
Don't think I've had occasion to write such? I'd just use `_connect` then.

Reply:
That was referring to newline before comment.

Reply:
Oh i do put a blank line below the `else:`.